spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/CachingConnectionFactory.java

```diff
@@ -801,6 +801,9 @@ private void logicalClose(ChannelProxy proxy) throws Exception {
 			if (this.target != null && !this.target.isOpen()) {
 				synchronized (targetMonitor) {
 					if (this.target != null && !this.target.isOpen()) {
+						if (this.target instanceof PublisherCallbackChannel) {
+							this.target.close(); // emit nacks if necessary
```

Comment:
Yes; I couldn't figure out why a test was failing with the changes I made and I found it was because an `isOpen()` returned false here instead of above.

I should really try to understand what changed. I'll take a look next week (if my old body survives shoveling 600mm + snow :frowning: ).